projects/packages/forms/src/contact-form/class-contact-form-endpoint.php

```diff
@@ -9,12 +9,406 @@
+ * Used as 'rest_controller_class' parameter when 'feedback' post type is
+ * registered in \Automattic\Jetpack\Forms\ContactForm\Contact_Form.
  */
 class Contact_Form_Endpoint extends \WP_REST_Posts_Controller {
+	// Check Akismet filters there... (contact_form_akismet)
+
+	// TODO: check about `get_responses_permission_check` differences...
+
+	/**
+	 * Registers the REST routes.
+	 *
+	 * @access public
+	 */
+	public function register_routes() {
+		parent::register_routes();
+		register_rest_route(
+			$this->namespace,
+			$this->rest_base . '/filters',
+			array(
+				'methods'             => \WP_REST_Server::READABLE,
+				'callback'            => array( $this, 'get_filters' ),
+				'permission_callback' => array( $this, 'get_items_permissions_check' ),
+			)
+		);
+		// TODO: add `args` to the route.
+		register_rest_route(
+			$this->namespace,
+			$this->rest_base . '/bulk_actions',
+			array(
+				'methods'             => \WP_REST_Server::CREATABLE,
+				'callback'            => array( $this, 'bulk_actions' ),
+				'permission_callback' => array( $this, 'get_items_permissions_check' ),
+			)
+		);
+	}
+
+	/**
+	 * Retrieves all distinct sources (posts) and all the distinct available dates that
+	 * any feedback was received, in order to be used as filters in the list.
+	 *
+	 * @return WP_REST_Response Response object on success.
+	 */
+	public function get_filters() {
+		// TODO: investigate how we can do this better regarding usage of $wpdb,
+		// performance by querying all the entities, etc..
+		global $wpdb;
+		// phpcs:disable WordPress.DB.DirectDatabaseQuery.DirectQuery,WordPress.DB.DirectDatabaseQuery.NoCaching,WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+		$months = $wpdb->get_results(
+			"SELECT DISTINCT YEAR( post_date ) AS year, MONTH( post_date ) AS month
+			FROM $wpdb->posts
+			WHERE post_type = 'feedback'
+			ORDER BY post_date DESC"
+		);
+		// phpcs:enable
+		$source_ids = Contact_Form_Plugin::get_all_parent_post_ids(
+			array_diff_key( array( 'post_status' => array( 'draft', 'publish', 'spam', 'trash' ) ), array( 'post_parent' => '' ) )
+		);
+		return rest_ensure_response(
+			array(
+				'date'   => array_map(
+					static function ( $row ) {
+						return array(
+							'month' => (int) $row->month,
+							'year'  => (int) $row->year,
+						);
+					},
+					$months
+				),
+				'source' => array_map(
+					static function ( $post_id ) {
+						return array(
+							'id'    => $post_id,
+							'title' => get_the_title( $post_id ),
+							'url'   => get_permalink( $post_id ),
+						);
+					},
+					$source_ids
+				),
+			)
+		);
+	}
+
+	/**
+	 * Adds the additional fields to the item's schema.
+	 *
+	 * @return array Item schema as an array.
+	 */
+	public function get_item_schema() {
+		$schema = parent::get_item_schema();
+
+		$schema['properties']['uid'] = array(
+			'description' => __( 'The UID... Updated description.', 'jetpack-forms' ),
```

Comment:
@Automattic/zap I was wondering if you could help with some of these missing descriptions.

Reply:
[pushed improved descriptions](https://github.com/Automattic/jetpack/pull/41602/commits/e41f64dcfd94d4962580a235089f704bf1a28d65)